### PR TITLE
Coverage

### PR DESCRIPTION
**Pull request by alyssawilk**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]
Fixes https://github.com/envoyproxy/envoy/issues/12009

### CODE REVIEW

**Review by RyanTheOptimist — APPROVED** [review]
Draft, but LGTM!

File: source/extensions/network/dns_resolver/getaddrinfo/getaddrinfo.cc [hunk 1]
```diff
@@ -102,29 +102,25 @@ class GetAddrInfoDnsResolver : public DnsResolver, public Logger::Loggable<Logge
     }
 
     std::list<DnsResponse> final_results;
-    switch (query.dns_lookup_family_) {
+    switch (query.dns_lookup_family_)
     case DnsLookupFamily::All: {
```
Comment (mergeconflict) [inline source/extensions/network/dns_resolver/getaddrinfo/getaddrinfo.cc:106]: Drive-by comment from Google's envoy security review: The opening brace here after the first case label, instead of immediately after the `switch (...)`, is surprising! I didn't know C++ allowed this. It seems to work fine, but you might want to move it up a line to avoid confusion.
